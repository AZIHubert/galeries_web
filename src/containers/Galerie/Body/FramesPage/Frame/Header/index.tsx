import * as React from 'react';
import { AiOutlineDelete } from 'react-icons/ai';
import { useSelector } from 'react-redux';

import Image from '#components/Image';
import Text from '#components/Text';

import theme from '#helpers/theme';

import {
  selectProfilPicture,
} from '#store/helpers';
import {
  userSelector,
} from '#store/selectors';

import {
  Container,
  DeleteButton,
  ProfilePictureImage,
  UserContainer,
} from './styles';

interface HeaderI {
  user: UserI;
}

const Header = ({
  user,
}: HeaderI) => {
  const currentUser = useSelector(userSelector);

  return (
    <Container>
      <UserContainer>
        <ProfilePictureImage>
          <Image
            alt={`${user.userName}'s profile picture`}
            mode='fill'
            original={selectProfilPicture(user).croped}
            pending={selectProfilPicture(user).pending}
          />
        </ProfilePictureImage>
        <Text
          fontWeight='bold'
          styles={{
            fontSize: 0.8,
          }}
        >
          {user.userName}
        </Text>
      </UserContainer>
      {
        currentUser
        && currentUser.id === user.id
        && (
          <DeleteButton>
            <AiOutlineDelete
              color={theme.colors.danger}
              size={15}
            />
          </DeleteButton>
        )
      }
    </Container>
  );
};

export default Header;
