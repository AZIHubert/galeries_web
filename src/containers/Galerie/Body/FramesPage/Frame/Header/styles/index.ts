export { default as Container } from './Container';
export { default as DeleteButton } from './DeleteButton';
export { default as ProfilePictureImage } from './ProfilePictureContainer';
export { default as UserContainer } from './UserContainer';
