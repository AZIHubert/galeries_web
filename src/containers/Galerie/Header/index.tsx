import * as React from 'react';

import Button from '#components/Button';
import Text from '#components/Text';

import {
  Container,
  CoverPicture,
  TitleContainer,
} from './styles';

interface HeaderI {
  galerie: GalerieI;
}

const Header = ({
  galerie,
}: HeaderI) => (
  <Container>
    <CoverPicture
      backgroundColor={galerie.defaultCoverPicture}
    >

    </CoverPicture>
    <TitleContainer>
      <div>
        <Text
          fontWeight='bold'
          styles={{
            fontSize: 1.9,
            lineHeight: 2.3,
          }}
        >
          {galerie.name}
        </Text>
        <Text>
          {galerie.users.length} member{galerie.users.length > 1 ? 's' : ''}
        </Text>
      </div>
      {galerie.role !== 'user' ? (
        <div>
          <Button.Default
            variant='secondary'
            title='create an invitation'
          />
        </div>
      ) : null}
    </TitleContainer>
  </Container>
);

export default Header;
