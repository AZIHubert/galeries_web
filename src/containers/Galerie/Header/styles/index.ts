export { default as Container } from './Container';
export { default as CoverPicture } from './CoverPicture';
export { default as TitleContainer } from './TitleContainer';
