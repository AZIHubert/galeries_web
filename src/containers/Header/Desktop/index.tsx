import * as React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  useHistory,
  useLocation,
} from 'react-router-dom';
import { CSSTransition } from 'react-transition-group';

import Button from '#components/Button';
import Modal from '#components/Modal';

import logo from '#ressources/svg/logoG.svg';
import {
  CreateGalerie,
  CreateGalerieHover,
  Home,
  HomeHover,
  Notification,
  NotificationHover,
} from '#ressources/svgComponents';

import { fetchLogout } from '#store/actions';
import { userSelector } from '#store/selectors';

import CreateGalerieModal from './CreateGalerieModal';
import Pictogram from './Pictogram';
import ProfileButton from './ProfileButton';
import {
  Container,
  Fader,
  HeaderPart,
  InnerContainer,
  Logo,
  StyledLink,
} from './styles';

const Header = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const user = useSelector(userSelector);
  const location = useLocation();
  const handleClickHome = () => history.push('/');
  const [openCreateGalerie, setOpenCreateGalerie] = React.useState<boolean>(false);

  const handleOpenCreateGalerie = () => setOpenCreateGalerie(true);
  const handleCloseCreateGalerie = () => setOpenCreateGalerie(false);

  const show = !!user
    && !location.pathname.includes('profilePicture')
    && !location.pathname.includes('updateEmail')
    && !location.pathname.includes('validateEmail');

  return (
    <CSSTransition
      classNames='fade'
      in={show}
      timeout={300}
      unmountOnExit
    >
      <Fader>
        <Container>
          <InnerContainer>
            <HeaderPart>
              <StyledLink
                to='/'
              >
                <Logo
                  alt="header logo"
                  src={logo}
                />
              </StyledLink>
            </HeaderPart>
            <HeaderPart>
              <Pictogram
                hoverPictogram={HomeHover}
                marginRight={25}
                marginRightL={35}
                onClick={handleClickHome}
                pictogram={Home}
              />
              <Pictogram
                hoverPictogram={CreateGalerieHover}
                marginRight={25}
                marginRightL={35}
                onClick={handleOpenCreateGalerie}
                pictogram={CreateGalerie}
              />
              <Pictogram
                hoverPictogram={NotificationHover}
                pictogram={Notification}
              />
              <ProfileButton />
              <Button.Header
                onClick={() => dispatch(fetchLogout())}
                small
                styles={{
                  marginLeft: 40,
                }}
                title='logout'
              />
            </HeaderPart>
          </InnerContainer>
          <Modal.Portal
            handleClose={handleCloseCreateGalerie}
            open={openCreateGalerie}
          >
            <CreateGalerieModal
              handleCloseCreateGalerie={handleCloseCreateGalerie}
            />
          </Modal.Portal>
        </Container>
      </Fader>
    </CSSTransition>
  );
};

export default Header;
