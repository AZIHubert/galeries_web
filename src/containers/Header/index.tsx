import * as React from 'react';
import styled from 'styled-components';

import mediaQueries from '#helpers/mediaQueries';

import HeaderButton from '#components/HeaderButton';

import createGalerie from '#ressources/svg/createGalerie.svg';
import home from '#ressources/svg/home.svg';
import logo from '#ressources/svg/logoG.svg';
import notification from '#ressources/svg/notification.svg';

import SearchBar from './SearchBar';
import ProfileButton from './ProfileButton';

const Container = styled.header`
  background-color: ${({ theme }) => theme.colors.secondary};
  position: absolute;
  width: 100%;
`;

const InnerContainer = styled.div`
  border-bottom: ${({ theme }) => (
    `2px solid ${theme.colors.primary}`
  )};
  display: flex;
  height: ${({ theme }) => `${theme.header.height.small}px`};
  justify-content: space-between;
  margin: ${({ theme }) => `0 ${theme.wrapper.margin.smallest}px`};
  @media ${mediaQueries.mobileL} {
    margin: ${({ theme }) => `0 ${theme.wrapper.margin.small}px`};
  }
  @media ${mediaQueries.laptop} {
    height: ${({ theme }) => `${theme.header.height.medium}px`};
    margin: ${({ theme }) => `0 ${theme.wrapper.margin.medium}px`};
  }
  @media ${mediaQueries.laptopL} { 
    height: ${({ theme }) => `${theme.header.height.large}px`};
    margin: ${({ theme }) => `0 ${theme.wrapper.margin.large}px`};
  }
`;

const Logo = styled.img`
  margin-right: 40px;
  width: ${({ theme }) => (
    `${theme.header.logoWidth.medium}px`
  )};
  @media ${mediaQueries.laptopL} { 
    width: ${({ theme }) => (
    `${theme.header.logoWidth.large}px`
  )};
  }
`;

const LeftHeader = styled.div`
  align-items: center;
  display: flex;
`;

const RightHeader = styled.div`
  align-items: center;
  display: flex;
`;

const Header = () => (
  <Container>
    <InnerContainer>
      <LeftHeader>
        <Logo
          src={logo}
          alt="header logo"
        />
        <SearchBar />
      </LeftHeader>
      <RightHeader>
        <button>
          <img
            alt='home pictogram'
            src={home}
          />
        </button>
        <button>
          <img
            alt='create new galerie pictogram'
            src={createGalerie}
          />
        </button>
        <button>
          <img
            alt='notification pictogram'
            src={notification}
          />
        </button>
        <ProfileButton />
        <HeaderButton
          title='logout'
        />
      </RightHeader>
    </InnerContainer>
  </Container>
);

export default Header;
