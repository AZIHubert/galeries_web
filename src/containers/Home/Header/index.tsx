import * as React from 'react';

import Modal from '#components/Modal';
import logo from '#ressources/svg/logoG.svg';

import ModalForgotPassword from './ModalForgotPassword';
import ModalLogin from './ModalLogin';
import ModalSignin from './ModalSignin';
import ModalVerifyAccount from './ModalVerifyAccount';
import ModalValidateResetPassword from './ModalValidateResetPassword';

const Header = () => {
  const [accountCreate, setAccountCreate] = React.useState<boolean>(false);
  const [forgotPassword, setForgotPassword] = React.useState<boolean>(false);
  const [loading, setLoading] = React.useState<boolean>(false);
  const [openLogin, setOpenLogin] = React.useState<boolean>(false);
  const [openSignin, setOpenSignin] = React.useState<boolean>(false);
  const [validateResetPassword, setValidateResetPassword] = React.useState<boolean>(false);
  const [currentEmail, setCurrentEmail] = React.useState<string>('');

  const handleClickLogin = () => {
    if (!loading) {
      setOpenSignin(false);
      setOpenLogin(!openLogin);
    }
  };
  const handleClickSignin = () => {
    if (!loading) {
      setOpenLogin(false);
      setOpenSignin(!openSignin);
    }
  };
  const handleCloseModal = () => {
    if (!loading) {
      setAccountCreate(false);
      setForgotPassword(false);
      setOpenLogin(false);
      setOpenSignin(false);
      setValidateResetPassword(false);
      setCurrentEmail('');
    }
  };
  const switchToValidateResetPassword = () => {
    if (!loading) {
      setForgotPassword(false);
      setValidateResetPassword(true);
    }
  };

  const LoggerModal = accountCreate ? (
    <ModalVerifyAccount
      currentEmail={currentEmail}
      loading={loading}
      setLoading={setLoading}
    />
  ) : (
    <ModalSignin
      loading={loading}
      setAccountCreate={setAccountCreate}
      setCurrentEmail={setCurrentEmail}
      setLoading={setLoading}
      switchModal={handleClickLogin}
    />
  );
  const SignerModal = () => {
    if (forgotPassword) {
      return (
        <ModalForgotPassword
          setCurrentEmail={setCurrentEmail}
          setForgotPassword={setForgotPassword}
          setLoading={setLoading}
          loading={loading}
          switchModal={switchToValidateResetPassword}
        />
      );
    }
    if (validateResetPassword) {
      return (
        <ModalValidateResetPassword
          currentEmail={currentEmail}
          loading={loading}
          setLoading={setLoading}
        />
      );
    }
    return (
      <ModalLogin
        loading={loading}
        setLoading={setLoading}
        setForgotPassword={setForgotPassword}
        switchModal={handleClickSignin}
      />
    );
  };

  return (
    <header>
      <img
        src={logo}
        alt="header logo"
      />
      <button
        onClick={handleClickSignin}
        data-testid='openSignin'
      >
        signin
      </button>
      <button
        data-testid='openLogin'
        onClick={handleClickLogin}
      >
        login
      </button>
      <Modal
        open={openLogin || openSignin}
        handleClose={handleCloseModal}
      >
        {openLogin && SignerModal()}
        {openSignin && LoggerModal}
      </Modal>
    </header>
  );
};

export default Header;
