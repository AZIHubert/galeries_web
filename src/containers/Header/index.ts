import Desktop from './Desktop';
import Home from './Home';

const Header = {
  Desktop,
  Home,
};

export default Header;
