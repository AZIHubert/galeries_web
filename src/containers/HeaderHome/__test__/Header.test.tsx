import {
  fireEvent,
  render,
  screen,
  act,
} from '@testing-library/react';
import * as React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import ThemeProvider from '#contexts/ThemeContext';

import Header from '../index';

describe('Header', () => {
  beforeAll(() => {
    // @ts-ignore
    ReactDOM.createPortal = jest.fn((element) => element);
  });
  beforeEach(() => {
    act(() => {
      render(
        <ThemeProvider>
          <Header />
        </ThemeProvider>,
      );
    });
  });
  afterEach(() => {
    // @ts-ignore
    ReactDOM.createPortal.mockClear();
  });
  it('renders without crashing', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Header />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render verify account modal if account is created', async () => {
    const email = 'user@email.com';
    const openSignin = screen.getByTestId('openSignin');
    fireEvent.click(openSignin);
    const confirmPasswordField = screen.getByTestId('confirmPasswordField');
    const emailField = screen.getByTestId('emailField');
    const passwordField = screen.getByTestId('passwordField');
    const submitButton = screen.getByTestId('submitButton');
    const userNameField = screen.getByTestId('userNameField');
    fireEvent.change(confirmPasswordField, { target: { value: 'Aaoudjiuvhds9!' } });
    fireEvent.change(emailField, { target: { value: email } });
    fireEvent.change(passwordField, { target: { value: 'Aaoudjiuvhds9!' } });
    fireEvent.change(userNameField, { target: { value: 'user' } });
    fireEvent.click(submitButton);
    await screen.findByTestId('modalVerifyAccount');
    const modalSignin = screen.queryByTestId('modalSignin');
    expect(modalSignin).toBeNull();
    const verifyAccountBody = screen.getByTestId('verifyAccountBody');
    expect(verifyAccountBody).toHaveTextContent(email);
  });
  it('should switch betwen login and forgotPassword modal', () => {
    const openLogin = screen.getByTestId('openLogin');
    fireEvent.click(openLogin);
    const forgotPasswordButton = screen.getByTestId('forgotPasswordButton');
    fireEvent.click(forgotPasswordButton);
    const modalForgotPassword = screen.queryByTestId('modalForgotPassword');
    expect(modalForgotPassword).not.toBeNull();
    const cancelButton = screen.getByTestId('cancelButton');
    fireEvent.click(cancelButton);
    const loginModal = screen.queryByTestId('loginModal');
    expect(loginModal).not.toBeNull();
  });
  it('should switch between login and signin modal', () => {
    const openLogin = screen.getByTestId('openLogin');
    fireEvent.click(openLogin);
    const switchToSignin = screen.getByTestId('switchToSignin');
    fireEvent.click(switchToSignin);
    const modalSignin = screen.queryByTestId('modalSignin');
    expect(modalSignin).not.toBeNull();
    const switchToLogin = screen.getByTestId('switchToLogin');
    fireEvent.click(switchToLogin);
    const loginModal = screen.queryByTestId('loginModal');
    expect(loginModal).not.toBeNull();
  });
  it('should switch to ModalValidateResetPassword if ValidateResetPassword form is valid', async () => {
    const email = 'user@email.com';
    const openLogin = screen.getByTestId('openLogin');
    fireEvent.click(openLogin);
    const forgotPasswordButton = screen.getByTestId('forgotPasswordButton');
    fireEvent.click(forgotPasswordButton);
    const emailField = screen.getByTestId('emailField');
    const submitButton = screen.getByTestId('submitButton');
    fireEvent.change(emailField, { target: { value: email } });
    fireEvent.click(submitButton);
    await screen.findByTestId('modalValidateResetPassword');
    const modalForgotPassword = screen.queryByTestId('modalForgotPassword');
    expect(modalForgotPassword).toBeNull();
    const validateResetPasswordBody = screen.getByTestId('validateResetPasswordBody');
    expect(validateResetPasswordBody).toHaveTextContent(email);
  });
});
