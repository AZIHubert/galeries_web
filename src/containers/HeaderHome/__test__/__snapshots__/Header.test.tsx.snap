// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<header
  className="sc-dmlrTW dFYIiB"
>
  <div
    className="sc-kfzAmx fRViZQ"
  >
    <img
      alt="header logo"
      className="sc-fKFyDc ZYkwf"
      src={Object {}}
    />
    <div
      className="sc-hHftDr kLgjWa"
    >
      <button
        className="sc-bdfBwQ kjBTJr"
        data-testid="openSignin"
        onClick={[Function]}
      >
        Sign in
      </button>
      <button
        className="sc-bdfBwQ ctZNxY"
        data-testid="openLogin"
        onClick={[Function]}
      >
        Log in
      </button>
    </div>
  </div>
</header>
`;
